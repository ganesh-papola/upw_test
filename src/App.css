

p,
h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0 !important;
}

input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
input[type="number"] {
  -moz-appearance: textfield;
}
