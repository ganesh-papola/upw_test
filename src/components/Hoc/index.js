export { default as LoaderHoc } from "./LoaderHoc";
