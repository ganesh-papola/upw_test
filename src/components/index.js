export * from "./common";
export * from "./Hoc";
