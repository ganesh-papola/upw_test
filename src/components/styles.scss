@import "../shared/styles/index.scss";

$max-zindex: 1000;
$mid-zindex: 999;
