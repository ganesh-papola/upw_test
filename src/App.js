import "./App.css";
import { Provider } from "react-redux";
import { persistor, store } from "./redux/store";
import Routes from "./routes/routes";
import { PersistGate } from "redux-persist/integration/react";
import { HelmetProvider } from "react-helmet-async";
import { LoaderHoc } from "./components";

function App() {
  return (
    <Provider store={store}>
      <PersistGate persistor={persistor}>
        <HelmetProvider>
          <LoaderHoc>
            <Routes />
          </LoaderHoc>
        </HelmetProvider>
      </PersistGate>
    </Provider>
  );
}

export default App;
