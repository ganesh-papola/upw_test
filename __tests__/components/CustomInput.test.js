import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { CustomInput } from "../../src/components"

describe('CustomInput test', () => {
    const props = {
        id: "testId123",
        label: "test",
        name: "test",
        errors: {},
        touched: {},
        handleChange: jest.fn(),
        containerClass: "",
        inputClass: "",
        labelClass: "",
        placeholder: "input",
        type: "text",
        value: "Hii",
    }
    it('this is components snapshot rendering test', () => {
        const { container } = render(
            <CustomInput {...props} />
        );
        expect(container).toMatchSnapshot();
    });
    it('this is components snapshot rendering test with error', () => {
        const newProps = {
            ...props,
            errors : {
                name: "some custom error",
            }
        }
        const { container } = render(
            <CustomInput {...props} />
        );
        expect(container).toMatchSnapshot();
    });
    it('should change the value while input', () => {
        const { queryAllByPlaceholderText } = render(<CustomInput {...props} />);
        const input = queryAllByPlaceholderText("input")[0];
        fireEvent.change(input, {target: {value: 'Test'}})
        expect(input.value).toBe('Hii')
    })
})
