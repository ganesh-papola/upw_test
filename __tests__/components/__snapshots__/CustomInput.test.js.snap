// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton test this is components snapshot rendering test 1`] = `
<div>
  <div
    class="inputContainer"
  >
    <div
      class="formGroup"
    >
      <input
        class="formField"
        id="testId123"
        max="100"
        min="10"
        name="test"
        placeholder="input"
        type="text"
        value="Hii"
      />
    </div>
    <div
      class="inputContainerLabelDiv"
    >
      <span
        class="inputContainerError"
      />
    </div>
  </div>
</div>
`;

exports[`CustomButton test this is components snapshot rendering test 2`] = `
<div>
  <div
    class="inputContainer"
  >
    <div
      class="formGroup"
    >
      <input
        class="formField"
        id="testId123"
        max="100"
        min="10"
        name="test"
        placeholder="input"
        type="text"
        value="Hii"
      />
    </div>
    <div
      class="inputContainerLabelDiv"
    >
      <span
        class="inputContainerError"
      />
    </div>
  </div>
</div>
`;

exports[`CustomButton test this is components snapshot rendering test with error 1`] = `
<div>
  <div
    class="inputContainer"
  >
    <div
      class="formGroup"
    >
      <input
        class="formField"
        id="testId123"
        max="100"
        min="10"
        name="test"
        placeholder="input"
        type="text"
        value="Hii"
      />
    </div>
    <div
      class="inputContainerLabelDiv"
    >
      <span
        class="inputContainerError"
      />
    </div>
  </div>
</div>
`;

exports[`CustomInput test this is components snapshot rendering test 1`] = `
<div>
  <div
    class="inputContainer"
  >
    <div
      class="formGroup"
    >
      <input
        class="formField"
        id="testId123"
        max="100"
        min="10"
        name="test"
        placeholder="input"
        type="text"
        value="Hii"
      />
    </div>
    <div
      class="inputContainerLabelDiv"
    >
      <span
        class="inputContainerError"
      />
    </div>
  </div>
</div>
`;

exports[`CustomInput test this is components snapshot rendering test with error 1`] = `
<div>
  <div
    class="inputContainer"
  >
    <div
      class="formGroup"
    >
      <input
        class="formField"
        id="testId123"
        max="100"
        min="10"
        name="test"
        placeholder="input"
        type="text"
        value="Hii"
      />
    </div>
    <div
      class="inputContainerLabelDiv"
    >
      <span
        class="inputContainerError"
      />
    </div>
  </div>
</div>
`;
